vault/resource_consul_secret_backend.go

```diff
@@ -226,6 +226,19 @@ func consulSecretBackendUpdate(d *schema.ResourceData, meta interface{}) error {
 
 	d.Partial(true)
 
+	if d.HasChange(consts.FieldPath) {
```

Comment:
Similar code seems to be repeated throughout, suggest generalizing this.

Reply:
Refactored to helper method `remountToNewPath`